src/crimson/net/SocketMessenger.h

```diff
@@ -152,6 +152,17 @@ class SocketMessenger final : public Messenger {
   Interceptor *interceptor = nullptr;
 #endif
 
+  void mark_down(const entity_addr_t& a) final {
+    auto conn = lookup_conn(a);
+    if (conn) {
+      std::ignore = seastar::smp::submit_to(
```

Comment:
Can this future be passed to the user?

It also reflects the fact that marking down a connection in another shard is not atomic.

Anyway, ignoring future isn't recommanded unless there is no simple way to handle it yet.

Reply:
ok, will do

Reply:
Done.

Reply:
BTW, I ignore the future return by `Messenger::mark_down()` in `OSD` when trying to mark down connections. I think it won't cause any trouble, am I right?

Reply:
See https://github.com/scylladb/seastar/blob/master/include/seastar/core/future.hh#L1210, it isn't recommended.

And probably in an extreme case, OSD shutdown might be affected, that OSD&Messenger is destructed before conn->mark_down() is called -- looks there is no guarantee if the future is ignored.

Gate it or chain it might be a choice.

Reply:
Ah, yes, will correct it.